src/pypromice/L0toL1.py

```diff
@@ -137,22 +139,31 @@ def _flagNAN(ds, flag_file=None):
     ds : xr.Dataset
         Level 0 data with flagged data
     '''
-    # flag_file = "./data/flags/" + ds.attrs["station_id"] + ".csv"
-    if flag_file is None:
-        print('Flag file not given. No data flagged')
+    ds = ds_in.copy()
+    try:
+        os.mkdir('local')
+        os.mkdir('local/flags')
+        os.mkdir('local/adjustments')
+    except:
```

Comment:
In general, I like to have the exception be specific to the expected error that is being handled, such as `except FileExistsError` if the directory already exists.

Reply:
Also, you could consider instead using `os.mkdirs`. https://docs.python.org/3/library/os.html#os.makedirs

This can create a tree of directories in one call. And you can also set `exist_ok=True` and it will not raise errors if the directory path already exists. Then you could just run:

```
        os.mkdirs('local/flags', exist_ok=True)
        os.mkdirs('local/adjustments', exist_ok=True)
```

Reply:
> In general, I like to have the exception be specific to the expected error that is being handled, such as except FileExistsError if the directory already exists. Then, if some other unhandled exception occurs that we are not expecting, we will see it. Maybe unlikely in such a narrow piece of code, but still good practice.

Like this?
https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/861067805bfe84dc348a2838dbbe7249ae523bb0/src/pypromice/L0toL1.py#L153

Reply:
> This can create a tree of directories in one call. And you can also set exist_ok=True and it will not raise errors if the directory path already exists. 

I've changed this in the new commit to assume that the user has cloned the [PROMICE-AWS-data-issues](https://github.com/GEUS-Glaciology-and-Climate/PROMICE-AWS-data-issues). If the local copy of this file doesn't exist in the file path as defined on the repo then it is skipped over. Good solution for now?

Reply:
Hey Penny, 
I am not convinced.
I think pypromice should fetch the latest csv files everytime it is being run, or at least try.
Otherwise it is up to us to manually run a git pull before re-running pypromice. I find it quite tedious.

Reply:
Okay. Currently these are the steps that occur:
1. Try to fetch .csv file from the Github repo (i.e. the most up-to-date version)
2. If fails to fetch from repo, try to fetch from local copy of repo
      a. If a .csv file exists, proceed with flagging/adjusting
      b. If a .csv file doesn't exist in either space, then skip flagging/adjusting

Do you want a step in there that tries to pull the latest repo updates to the local copy?

Reply:
Yes, I was suggesting earlier that we should not depend on the user cloning the data-issues repo. But it seems like that is not the case with @PennyHow 's latest changes.

@BaptisteVandecrux 's original implementation had the advantage that it would write from the URL source to the `local` directory on each call. In these latest changes, there is no way to incrementally update the local files.

Reply:
We could have an option download from URL if a user requires it. I could add a flag to the `_getDF` function that if `True` downloads and saves the `.csv` from URL.

Reply:
Fine with a download csv option, even more if we can we have it as `True` as default

Reply:
As implemented, if the processing is being run locally many times, and you have some local directory path for the csv flag files.... The local files would never update (they only update if the URL is not available). Then if you wanted to run offline, you would fall back to using potentially very old csv files. Whereas, Baptiste's original implementation (with the `urllib.request.urlretrieve()`) would write to the local directory everytime, essentially synching it to the URL source. This is kind of like having a repo being updated, but removes the requirement of a user having to clone and update another repo  for manual processing.

Reply:
Of course, it seems like the internet is everywhere all the time these days, so maybe we don't need to worry about losing the URL source too much :)

Reply:
New changes include a download option for the flag and adjustment files.

https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/2bc410319f640e30edfdfd31f89fbff5b8d40013/src/pypromice/L0toL1.py#L144-L154

So if `download` is `True` and the directory for the flag/adjustment files exists then it will make a local copy.

I also included the `pandas` version specification that @BaptisteVandecrux raised in #93. 

The only thing is, I get a deprecation warning with `xarray` though, related to the time shift correction (when running the `KPC_Uv3` adjustment).

```
/home/pho/anaconda3/envs/pypromice/lib/python3.8/site-packages/xarray/core/indexes.py:150: FutureWarning: Indexing a timezone-naive DatetimeIndex with a timezone-aware datetime is deprecated and will raise KeyError in a future version.  Use a timezone-naive object instead.
  indexer = index.slice_indexer(
  ```

Reply:
New change points downloads to `loca/flags` and `local/adjustments` directories.

Reply:
> Also, you could consider instead using `os.mkdirs`. https://docs.python.org/3/library/os.html#os.makedirs
> 
> This can create a tree of directories in one call. And you can also set `exist_ok=True` and it will not raise errors if the directory path already exists. Then you could just run:
> 
> ```
>         os.mkdirs('local/flags', exist_ok=True)
>         os.mkdirs('local/adjustments', exist_ok=True)
> ```

using your suggestion

Reply:
In the latest changes, the read_csv from url was removed, it made local debugging and local construction of flags and adjustments files very tedious. We are now back to something close to the original formulation with:

If download = True, it  urllib.request.urlretrieve the flags and adjustments files on the data-issue github repo, stores it in the local folder, replacing any existing file. 

If download = False or urlretrieve failed, the local files are not updated.

The code then reads the local flags and adjustments files if any.